samlang-core/src/services/__tests__/source-processor.test.ts

```diff
@@ -64,3 +86,108 @@ GLOBAL_STRING_1:
     .text
 `);
 });
+
+it('compile hello world to JS integration test', () => {
+  const moduleReference = new ModuleReference(['Test']);
+  const sourceCode = `
+    class Main {
+        function main(): unit = println('Hello '::'World!')
```

Comment:
There is an syntax error here: samlang strings use double-quote. The syntax error causes the AST node to be dropped, that's why you see some garbage in JS output.
Classic garbage-in-garbage-out example :)